Dockerfile

```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM golang:1.20 as builder
+FROM golang:1.21 as builder
```

Comment:
I guess that would've also taken care of [this](https://github.com/numaproj-labs/numaplane/pull/149) issue, right? So, now we should be good to use go 1.21 as long as we all are running that version locally